src/liger_kernel/chunked_loss/orpo_loss.py

```diff
@@ -0,0 +1,180 @@
+import torch
+from triton import next_power_of_2
+import torch.nn.functional as F
+from liger_kernel.ops.utils import element_mul_kernel
+
+# The hard limit of TRITON_MAX_TENSOR_NUMEL is 1048576 https://github.com/triton-lang/triton/blob/ba42a5c68fd0505f8c42f4202d53be0f8d9a5fe0/python/triton/language/core.py#L19
+# However, setting limit as 65536 as in LayerNorm tutorial is faster because of less register spilling
+# The optimal maximum block size depends on your hardware, your kernel, and your dtype
+MAX_FUSED_SIZE = 65536 // 2
+
+
+def odds_ratio_loss(chosen_logps, rejected_logps, beta=0.1):
+    """
+    Compute odds-ratio loss.
+    Args:
+        chosen_logps (torch.Tensor): Avg log probabilities of chosen tokens. Shape: (batch_size,).
+        rejected_logps (torch.Tensor): Avg log probabilities of rejected tokens. Shape: (batch_size,).
+        beta (float): Weight for the odds ratio loss.
+    """
+    log_odds = (chosen_logps - rejected_logps) - (
+        torch.log1p(-torch.exp(chosen_logps)) - torch.log1p(-torch.exp(rejected_logps))
+    )
+    ratio = F.logsigmoid(log_odds)
+    return beta * ratio.sum()
+
+
+class LigerFusedLinearORPOFunction(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, _input, weight, target, bias=None, ignore_index=-100, beta=0.1, compiled=True):
+        """
+        Fused linear forward function with ORPO (Odds-Ratio Preference Optimization).
```

Comment:
Can you please add a reference link to the paper or a blog explaining the function.